server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueFactory.java

```diff
@@ -77,7 +80,11 @@ public void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueExcept
 
         @Override
         public void enQueue(Mail mail) throws MailQueueException {
-            mailItems.addFirst(new MemoryMailQueueItem(mail));
+            try {
+                mailItems.addFirst(new MemoryMailQueueItem(new MailImpl(mail)));
```

Comment:
The mail passed by the argument of the method will be disposed by the caller
The mail enqueued will be disposed by the caller of dequeue

Is that more clear ?